### PR TITLE
refactor: replace empty() Part 1

### PR DESCRIPTION
**Pull request by kenjis**
~~Needs #8344~~

**Description**
- replace empty()

**Checklist:**
- [x] Securely signed commits
- [x] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: system/Model.php [hunk 1]
```diff
@@ -257,13 +257,13 @@ protected function doFirst()
 
         if ($this->tempUseSoftDeletes) {
             $builder->where($this->table . '.' . $this->deletedField, null);
-        } elseif ($this->useSoftDeletes && empty($builder->QBGroupBy) && $this->primaryKey) {
+        } elseif ($this->useSoftDeletes && ($builder->QBGroupBy === []) && $this->primaryKey) {
```
Comment (michalsn) [inline system/Model.php:260]: Is `()` around `$builder->QBGroupBy === []`  is forced by our tooling? Seems unnecessary.

File: system/Model.php [hunk 2]
```diff
@@ -257,13 +257,13 @@ protected function doFirst()
...
             $builder->groupBy($this->table . '.' . $this->primaryKey);
         }
 
         // Some databases, like PostgreSQL, need order
         // information to consistently return correct results.
-        if ($builder->QBGroupBy && empty($builder->QBOrderBy) && $this->primaryKey) {
+        if ($builder->QBGroupBy && ($builder->QBOrderBy === []) && $this->primaryKey) {
```
Comment (michalsn) [inline system/Model.php:266]: Same as above.

File: system/Test/FeatureTestCase.php [hunk 3]
```diff
@@ -186,7 +186,7 @@ public function call(string $method, string $path, ?array $params = null)
             ->run($routes, true);
 
         $output = \ob_get_contents();
-        if (empty($response->getBody()) && ! ($output === '' || $output === false)) {
+        if (($response->getBody() === null) && ! ($output === '' || $output === false)) {
```
Comment (michalsn) [inline system/Test/FeatureTestCase.php:189]: Probably unnecessary `()` around `$response->getBody() === null`.

File: system/Validation/Validation.php [hunk 4]
```diff
@@ -918,7 +918,7 @@ protected function getErrorMessage(
         $message = str_replace('{field}', ($label === null || $label === '') ? $field : lang($label), $message);
         $message = str_replace(
             '{param}',
-            empty($this->rules[$param]['label']) ? $param : lang($this->rules[$param]['label']),
+            (! isset($this->rules[$param]['label'])) ? $param : lang($this->rules[$param]['label']),
```
Comment (michalsn) [inline system/Validation/Validation.php:921]: Do we need `()` around `! isset($this->rules[$param]['label'])`?

File: system/View/Parser.php [hunk 5]
```diff
@@ -103,7 +103,7 @@ public function render(string $view, ?array $options = null, ?bool $saveData = n
         }
 
         $fileExt = pathinfo($view, PATHINFO_EXTENSION);
-        $view    = empty($fileExt) ? $view . '.php' : $view; // allow Views as .html, .tpl, etc (from CI3)
+        $view    = ($fileExt === '') ? $view . '.php' : $view; // allow Views as .html, .tpl, etc (from CI3)
```
Comment (michalsn) [inline system/View/Parser.php:106]: Same here... is it supposed to increase readability? For me, it's the opposite.

File: system/View/Parser.php [hunk 6]
```diff
@@ -589,7 +589,7 @@ protected function applyFilters(string $replace, array $filters): string
             preg_match('/\([\w<>=\/\\\,:.\-\s\+]+\)/u', $filter, $param);
 
             // Remove the () and spaces to we have just the parameter left
-            $param = ! empty($param) ? trim($param[0], '() ') : null;
+            $param = ($param !== []) ? trim($param[0], '() ') : null;
```
Comment (michalsn) [inline system/View/Parser.php:592]: With only one check, I see no added value in using `()`.

File: system/View/View.php [hunk 7]
```diff
@@ -178,7 +178,7 @@ public function render(string $view, ?array $options = null, ?bool $saveData = n
 
         $fileExt = pathinfo($view, PATHINFO_EXTENSION);
         // allow Views as .html, .tpl, etc (from CI3)
-        $this->renderVars['view'] = empty($fileExt) ? $view . '.php' : $view;
+        $this->renderVars['view'] = ($fileExt === '') ? $view . '.php' : $view;
```
Comment (michalsn) [inline system/View/View.php:181]: Same here

File: system/View/View.php [hunk 8]
```diff
@@ -207,12 +207,12 @@ public function render(string $view, ?array $options = null, ?bool $saveData = n
             $this->renderVars['file'] = $this->loader->locateFile(
                 $this->renderVars['view'],
                 'Views',
-                empty($fileExt) ? 'php' : $fileExt
+                ($fileExt === '') ? 'php' : $fileExt
```
Comment (michalsn) [inline system/View/View.php:210]: And here

**Review by michalsn — APPROVED** [review]
I left a few comments, but it may be only my preference.

File: system/View/Parser.php (hunk 5 above)
Comment (kenjis) [reply system/View/Parser.php:106]: In this case, yes.

**Comment by kenjis** [comment]
There are some unnecessary `()`, but I merge this for now.

**Pull request merged**